src/lib/tvgPaint.cpp

```diff
@@ -77,6 +77,9 @@ static bool _clipPathFastTrack(Paint* cmpTarget, const RenderTransform* pTransfo
             y2 = y2 * pTransform->m.e22 + pTransform->m.e23;
         }
 
+        if (x1 < 0.0f) x1 = 0.0f;
+        if (y1 < 0.0f) y1 = 0.0f;
```

Comment:
I think that the problem is a bit deeper. for now we don't use the viewBox attribute exactly as we should. please check the w3.org for more details. the viewBox can scale the svg (viewBox + width & height attributes) or shift it (x and y values of the viewBox attribute != 0). it doesn't work this way in tvg for now. 

So the problem is not only for x,y < 0, but for values > 0 we don't get the correct result as well.
By overriding the x1 and y1 values by 0 we lose some information that we didn't use yet.

Reply:
For this patch we don't lose information. The region is just for a fastTrack and it concerns ClipPath in general, not only SVG.
The issue that you mention should be taken separately. Is there some visible difference between tvg and validated viewer? Could you post such an svg file and create an issue for it?

Reply:
@mmaciola
"Could you post such an svg file and create an issue for it?"   ->   first thing tomorrow :)